### PR TITLE
emburse email engagements log initial built

### PR DESCRIPTION
**Pull request by hafizasyraf72**

### CODE REVIEW

**Comment by izztsfn** [comment]
Also @hafizasyraf72 i believe email_engagements_log script should be placed in Normalization folder instead of Ads Performance. Please create a separate commit for this ya. Can ping me if you're unsure about this thanksss

**Comment by hafizasyraf72** [comment]
ohhh yaaa. my bad. thanks for the checking @izztsfn. I have replace it to the other folder named as "Normalization Script"

File: Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+  --WHERE primary_attribute_value LIKE '2024-NN-EMB-APAC-MM-ENT-Short-Term-Disposition-Cold-Opp.Email #1'
+  QUALIFY ROW_NUMBER() OVER(PARTITION BY leadid, primary_attribute_value_id ORDER BY activitydate DESC) = 1
+),
+open_click AS ( --merge open and click data
+    SELECT * FROM email_open
+    UNION ALL
+    SELECT * FROM email_click
+),
+new_open AS ( --to populate the data in Clicked but not appear in Opened list
+    SELECT _sdc_sequence,
```
Comment (izztsfn) [inline Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql:178]: @hafizasyraf72 indent the field under SELECT clause

File: Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+           _campaign,
+           _subject,
+           _email,
+           _timestamp,
+           'Opened' AS _engagement,
+           _description,
+           _leadid,
+           _link 
+    FROM open_click
+    WHERE _engagement <> 'Opened' AND _engagement = 'Clicked'
```
Comment (izztsfn) [inline Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql:189]: @hafizasyraf72 add new line for AND and ensure it is inline with the first condition

File: Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+  UNION ALL
+  SELECT * FROM email_click
+  UNION ALL
+  SELECT * FROM email_hard_bounce
+  UNION ALL
+  SELECT * FROM email_soft_bounce
+  UNION ALL
+  SELECT * FROM email_unsubscribed
+)
+  SELECT
```
Comment (izztsfn) [inline Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql:289]: @hafizasyraf72 reduce one indentation for the entire portion (SELECT onwards until end of the file)

**Review by izztsfn — CHANGES_REQUESTED** [review]

File: Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+  UNION ALL
+  SELECT * FROM email_soft_bounce
+),
+new_delivered_email AS( --remove soft and hard bounced in delivered list
+    SELECT d.*
+    FROM email_delivered d
+    LEFT JOIN email_soft_hard_bounced b 
+      ON d._campaignID = b._campaignID 
+      AND d._leadid = b._leadid
+    WHERE b._campaignID IS NULL AND b._leadid IS NULL
```
Comment (raaayden) [inline Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql:240]: indent the AND on line 240

File: Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql (hunk 1 above)
Comment (hafizasyraf72) [reply Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql:178]: done indent the field under SELECT for each CTE
Screenshot 2024-09-24 103807

File: Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql (hunk 2 above)
Comment (hafizasyraf72) [reply Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql:189]: done amend this too

image

File: Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql (hunk 3 above)
Comment (hafizasyraf72) [reply Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql:289]: this query after final CTE also had been reduce one indentation
image

File: Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql (hunk 4 above)
Comment (hafizasyraf72) [reply Clients/Emburse/Normalization Script/emburse_db_email_engagement_log.sql:240]: done indent at WHERE statement
image

**Review by izztsfn — APPROVED** [review]
Looks good!

**Pull request merged**